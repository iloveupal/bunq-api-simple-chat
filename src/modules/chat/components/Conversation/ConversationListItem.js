import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
    getConversationUsers,
} from 'Modules/chat/domains/conversations/ConversationsPropGetters';

import {
    renderConversationName,
} from 'Modules/chat/domains/conversations/ConversationsUtils';

import UserSmallCircles from 'Modules/chat/components/User/UserSmallCircles';


const Container = styled.div`
    display: flex;
    height: 80px;
    width: 300px;
    background-color: #222;
    padding-left: 15px;
    padding-top: 15px;
    box-sizing: border-box;
    font-family: sans-serif;
    flex-flow: column nowrap;
    cursor: pointer;
    
    &:hover {
        background-color: #333;
    }
`;

const ConversationName = styled.span`
    font-weight: bold;
    font-size: 14px;
    color: white;
`;


export default class ConversationListItem extends PureComponent {
    static propTypes = {
        data: PropTypes.shape({
            users: PropTypes.array.isRequired,
            conversation: PropTypes.object.isRequired,
        }).isRequired,
        onClick: PropTypes.func.isRequired,
    };

    static HEIGHT = 80;

    handleClick = () => {
        this.props.onClick(this.props.data);
    };

    render () {
        return (
            <Container onClick={this.handleClick}>
                <ConversationName>
                    { renderConversationName(this.props.data) }
                </ConversationName>

                <UserSmallCircles
                    users={getConversationUsers(this.props.data)}
                />
            </Container>
        );
    }
}
