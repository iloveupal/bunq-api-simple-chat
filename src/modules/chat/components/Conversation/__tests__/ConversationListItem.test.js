import React from 'react';
import { shallow }  from 'enzyme';
import 'jest-styled-components';

import ConversationListItem from '../ConversationListItem';

describe('ConversationListItem', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(
            <ConversationListItem
                data={{ conversation: { conversationId: 1, name: 'Test' }, users: [{ userid: 1, }]}}
                onClick={jest.fn()}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
