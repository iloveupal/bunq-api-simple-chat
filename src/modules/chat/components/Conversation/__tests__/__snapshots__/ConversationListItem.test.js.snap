// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationListItem should match snapshot 1`] = `
<ConversationListItem__Container
  onClick={[Function]}
>
  <ConversationListItem__ConversationName>
    Test
  </ConversationListItem__ConversationName>
  <Connect(UserSmallCircles)
    users={
      Array [
        Object {
          "userid": 1,
        },
      ]
    }
  />
</ConversationListItem__Container>
`;
